core/sql/comexe/ComTdb.h

```diff
@@ -1143,6 +1145,33 @@ class ComTdbVirtTableSequenceInfo : public ComTdbVirtTableBase
   Int64                  redefTime;
 };
 
+
+// This class describes object and column privileges and if they are grantable 
+// (WGO) for an object. Privileges are stored as a vector of PrivMgrDesc's, one
+// per distinct grantee.  
+//
+//    PrivMgrDesc:
+//      grantee - Int32
+//      objectPrivs - PrivMgrCoreDesc 
+//      columnPrivs - list of PrivMgrCoreDesc 
+//    PrivMgrCoreDesc:
+//      bitmap of granted privileges
+//      bitmap of associated WGO (with grant option)
+//      column ordinal (number) set to -1 for object privs
+class ComTdbVirtTablePrivInfo : public ComTdbVirtTableBase
+{
+ public:
+  ComTdbVirtTablePrivInfo()
+    : ComTdbVirtTableBase()
+    {
+      init();
+    }
+
+  virtual Int32 size() { return sizeof(ComTdbVirtTablePrivInfo);}
+
+  std::vector<PrivMgrDesc>     *privmgr_desc_list;     
```

Comment:
I am not sure on all the memory rules and did wonder if I should include a vector class here. I could convert the std::vector of PrivMgrDesc to an NAList of PrivMgrDescs if that would be better.